src/frontend/src/handler/create_source.rs

```diff
@@ -654,25 +699,24 @@ pub fn bind_connector_props(
 /// how to handle the regular columns (i.e., non-generated) defined in SQL?
 pub enum SqlColumnStrategy {
     /// Follow all columns defined in SQL, ignore the columns from external system.
+    /// This ensures that no accidental side effect will change the schema.
     ///
-    /// This is the behavior when `SINK INTO` or `[ADD | DROP] COLUMN` atop the purified SQL,
-    /// ensuring that no accidental refresh will happen and the schema remains unchanged.
-    // TODO(purify): we may validate whether defined columns are valid against the resolved schema.
-    Follow,
+    /// This is the behavior when re-planning the target table of `SINK INTO`.
+    FollowUnchecked,
+
+    /// Follow all column defined in SQL, check the columns from external system with name & type.
+    /// This ensures that no accidental side effect will change the schema.
+    ///
```

Comment:
```suggestion
    ///
    /// - When wildcard （`*`) is presented, all columns are used, and the user-provided columns will be checked against the actual schema.
    /// - When wildcard is not presented, only columns from user's definition will be used and checked.
    ///
```

Reply:
This is not that true. When `*` is presented, regular (non-generated) columns specified by users will be rejected.

https://github.com/risingwavelabs/risingwave/blob/4cce72a149bccbea19aef2412d9b7e1cd2ffa33b/src/frontend/src/handler/create_source.rs#L239-L245

I will include this in the release notes part of this PR.